tiatoolbox/models/classification/patch_predictor.py

```diff
@@ -0,0 +1,382 @@
+            batch_output = self.__postprocess(batch_output_probs)
+            preds_output.extend(batch_output.tolist())
+            if return_probs:
+                # return raw output
+                probs_output.extend(batch_output_probs.tolist())
+            if return_labels:
+                labels_output.extend(batch_labels.tolist())
+
+            # may be a with block + flag would be nicer
+            if self.verbose:
+                pbar.update()
+        if self.verbose:
+            pbar.close()
+
+        preds_output = np.array(preds_output)
+
+        all_output = {"preds": preds_output}
+        if return_probs:
+            all_output["probs"] = probs_output
+        if return_labels:
+            all_output["labels"] = labels_output
+        return all_output
+
+
+def get_predefined_model(predefined_model=None, pretrained_weight=None):
+    """Load a predefined PyTorch model with the appropriate pretrained weights.
+
+    Args:
+        predefined_model (str): name of the existing models support by tiatoolbox
+            for processing the data. Currently supports:
+            - alexnet-kather100K: alexnet backbone trained on Kather 100K dataset.
+            - resnet18-kather100K: resnet18 backbone trained on Kather 100K dataset.
+            - resnet34-kather100K: resnet34 backbone trained on Kather 100K dataset.
+            - resnet50-kather100K: resnet50 backbone trained on Kather 100K dataset.
+            - resnet101-kather100K: resnet101 backbone trained on Kather 100K dataset.
+            - resnext5032x4d-kather100K: resnext50_32x4d backbone trained on Kather
+                100K dataset.
+            - resnext101_32x8d-kather100K: resnext101_32x8d backbone trained on
+                Kather 100K dataset.
+            - wide_resnet50_2-kather100K: wide_resnet50_2 backbone trained on
+                Kather 100K dataset.
+            - wide_resnet101_2-kather100K: wide_resnet101_2 backbone trained on
+                Kather 100K dataset.
+            - densenet121-kather100K: densenet121 backbone trained on
+                Kather 100K dataset.
+            - densenet161-kather100K: densenet161 backbone trained on
+                Kather 100K dataset.
+            - densenet169-kather100K: densenet169 backbone trained on
+                Kather 100K dataset.
+            - densenet201-kather100K: densenet201 backbone trained on
+                Kather 100K dataset.
+            - mobilenet_v2-kather100K: mobilenet_v2 backbone trained on
+                Kather 100K dataset.
+            - mobilenet_v3_large-kather100K: mobilenet_v3_large backbone trained on
+                Kather 100K dataset.
+            - mobilenet_v3_small-kather100K: mobilenet_v3_small backbone trained on
+                Kather 100K dataset.
+            - googlenet-kather100K: googlenet backbone trained on Kather 100K dataset.
+
+            By default, the corresponding pretrained weights will also be downloaded.
```

Comment:
This is not rendering quite right on sphinx/readthedocs. Please check spacing and indentation.